### PR TITLE
Add missing flags for some of the ENV vars

### LINKED ISSUE #4894: server: add missing flags for env vars
There are a few env vars which does not have corresponding flags are read out of the typical config reading path:

```
HASURA_GRAPHQL_EVENTS_HTTP_POOL_SIZE
HASURA_GRAPHQL_EVENTS_FETCH_INTERVAL
LOG_HEADERS_FROM_ENV
```

Also, `LOG_HEADERS_FROM_ENV` should be prefixed with `HASURA_GRAPHQL_`

### PR DESCRIPTION
**Pull request by codingkarthik**
fixes #4894 

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Server
- [ ] Console
- [ ] CLI
- [x] Docs
- [ ] Community Content
- [ ] Build System
- [ ] Tests
- [ ] Other (list it)

### Solution and Design

Add server flags for 

```
HASURA_GRAPHQL_EVENTS_HTTP_POOL_SIZE - events-http-pool-size
HASURA_GRAPHQL_EVENTS_FETCH_INTERVAL - events-fetch-interval
HASURA_GRAPHQL_LOG_HEADERS_FROM_ENV - log-headers-from-env
```

Also, added the prefix `HASURA_GRAPHQL_` to the `LOG_HEADERS_FROM_ENV` flag. 
The graphql-engine will first check if the flag `--log-headers-from-env` is set, if it's not then `HASURA_GRAPHQL_LOG_HEADERS_FROM_ENV` will be checked if that's also not set then finally `LOG_HEADERS_FROM_ENV` will be checked

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [x] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [x] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

**Comment by tirumaraiselvan** [comment]
I think we can get rid of `LOG_HEADERS_FROM_ENV` completely as

1) it was totally undocumented
2) it touches a very low surface area of usage
3) even if someone fished it out of the code-base, the worst that will happen is that they will see REDACTED values in their logs.

**Comment by codingkarthik** [comment]
Sure, I'll remove the `LOG_HEADERS_FROM_ENV` flag.

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ Read more about the session argument for computed fields in the [docs](https://h
 - server: fix importing of allow list query from metadata (fix #4687)
 - server: flush log buffer during shutdown (#4800)
 - server: fix edge case with printing logs on startup failure (fix #4772)
+- server: add flags for HASURA_GRAPHQL_EVENTS_HTTP_POOL_SIZE, HASURA_GRAPHQL_EVENTS_FETCH_INTERVAL and add a new config environment variable HASURA_GRAPHQL_LOG_HEADERS_FROM_ENV and a corresonding flag to it.
```
Comment (tirumaraiselvan) [inline CHANGELOG.md:75]: Can we give the names for the flags instead?

Something like "Added startup flags --a , --b , --c"

File: docs/graphql/manual/deployment/graphql-engine-flags/reference.rst [hunk 2]
```diff
@@ -128,14 +128,20 @@ For the ``serve`` sub-command these are the available flags and ENV variables:
...
      - ``HASURA_GRAPHQL_EVENTS_FETCH_INTERVAL``
      - Interval in milliseconds to sleep before trying to fetch events again after a fetch
-       returned no events from postgres
+       returned no events from postgres (default: 1000 ms)
+
+   * - ``--log-headers-from-env``
+     - ``HASURA_GRAPHQL_LOG_HEADERS_FROM_ENV``
+     - Flag to indicate if whether the values of the environment variables in the headers
+       should be logged. When set to false, the name of the environment variable will be
+       logged (default: false)
```
Comment (tirumaraiselvan) [inline docs/graphql/manual/deployment/graphql-engine-flags/reference.rst:144]:
```suggestion
     - Flag to indicate if environment variable should be expanded, if present in the log. When set to false, the name of the environment variable will be logged (default: false)
```

File: CHANGELOG.md (hunk 1 above)
Comment (codingkarthik) [reply CHANGELOG.md:75]: I'll update it.

File: docs/graphql/manual/deployment/graphql-engine-flags/reference.rst [hunk 3]
```diff
@@ -128,14 +128,20 @@ For the ``serve`` sub-command these are the available flags and ENV variables:
...
+   * - ``--events-fetch-interval <MILLISECONDS>``
      - ``HASURA_GRAPHQL_EVENTS_FETCH_INTERVAL``
      - Interval in milliseconds to sleep before trying to fetch events again after a fetch
-       returned no events from postgres
+       returned no events from postgres (default: 1000 ms)
+
+   * - ``--log-headers-from-env``
+     - ``HASURA_GRAPHQL_LOG_HEADERS_FROM_ENV``
+     - If an environment variable is present in a log line, this flag indicates
+       if it's value should be logged. When set to false, only the name of the
```
Comment (marionschleifer) [inline docs/graphql/manual/deployment/graphql-engine-flags/reference.rst:143]:
```suggestion
       if its value should be logged. When set to false, only the name of the
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -72,6 +72,8 @@ Read more about the session argument for computed fields in the [docs](https://h
 - server: fix importing of allow list query from metadata (fix #4687)
 - server: flush log buffer during shutdown (#4800)
 - server: fix edge case with printing logs on startup failure (fix #4772)
+- server: add startup flags --events-http-pool-size,--events-fetch-interval and --log-headers-from-env for corresponding
+  environment variables HASURA_GRAPHQL_EVENTS_HTTP_POOL_SIZE,HASURA_GRAPHQL_EVENTS_FETCH_INTERVAL and HASURA_GRAPHQL_LOG_HEADERS_FROM_ENV (fix #4894)
```
Comment (tirumaraiselvan) [inline CHANGELOG.md:76]:
```suggestion
- server: add startup flags --events-http-pool-size,--events-fetch-interval and --log-headers-from-env for corresponding event trigger configuration environment variables (fix #4894) (#4910)
```

**Comment by codingkarthik** [comment]
Not relevant anymore, as this information is already documented.

**Pull request closed without merging**